diff --git a/server/src/services/backup.service.ts b/server/src/services/backup.service.ts
index 1febe7913..3e03687d5 100644
--- a/server/src/services/backup.service.ts
+++ b/server/src/services/backup.service.ts
@@ -85,7 +85,9 @@ export class BackupService extends BaseService {
     } = this.configRepository.getEnv();
 
     const isUrlConnection = config.connectionType === 'url';
-    const databaseParams = isUrlConnection ? ['-d', config.url] : ['-U', config.username, '-h', config.host];
+    const databaseParams = isUrlConnection
+      ? ['-d', config.url]
+      : ['-U', config.username, '-h', config.host, '-p', config.port.toString()];
     const backupFilePath = path.join(
       StorageCore.getBaseFolder(StorageFolder.BACKUPS),
       `immich-db-backup-${Date.now()}.sql.gz.tmp`,
